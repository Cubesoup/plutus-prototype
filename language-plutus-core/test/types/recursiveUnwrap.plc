(program 0.1.0
  (abs a (type)
  (lam x
    (fix list (all r (type) (fun r (fun (fun a (fun list r)) r))))
    [
      [
        { (unwrap x) a }
        (error a)
      ]
      (lam head
        a
        (lam tail
          (fix list (all r (type) (fun a (fun (fun a (fun list a)) a))))
          head
        )
      )
    ]
  )
  )
)
