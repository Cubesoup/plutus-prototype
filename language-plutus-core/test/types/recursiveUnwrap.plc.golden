Type mismatch at 5:5 in term
  '(lam
    head
    a
    (lam
      tail (fix list (all r (type) (fun a (fun (fun a (fun list a)) a)))) head
    )
  )'.
Expected type
  '(fun a (fun (fix list (all r (type) (fun r (fun (fun a (fun list r)) r)))) a))',
found type
  '(fun a (fun (fix list (all r (type) (fun a (fun (fun a (fun list a)) a)))) a))'